### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Ros-an**
added component and code section for some

### CODE REVIEW

File: component/alert.css [hunk 1]
```diff
@@ -0,0 +1,27 @@
+.alert {
+  max-width: 50%;
+  min-width: 300px;
```
Comment (DanishOnGit) [inline component/alert.css:3]: Hi Roshan,
Avoid absolute units as far as possible.

File: component/alert.css [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  border-radius: 5px;
+  border: 1px solid rgba(0,0,0,.125);
+  font-size: 1rem;
+  position: relative;
+  background: white;
+  box-shadow: 3px 3px 10px 3px rgba(14, 14, 95,0.08);
+}
+.alert-hr {
+  border: none;
+  border-top: 1px solid rgba(0,0,0,.125);
```
Comment (DanishOnGit) [inline component/alert.css:15]: You can use a css variable for color here

File: component/alert.css [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  max-width: 50%;
+  min-width: 300px;
+  margin: auto;
+  padding: 0.75rem 1rem;
+  border-radius: 5px;
+  border: 1px solid rgba(0,0,0,.125);
+  font-size: 1rem;
+  position: relative;
+  background: white;
+  box-shadow: 3px 3px 10px 3px rgba(14, 14, 95,0.08);
```
Comment (DanishOnGit) [inline component/alert.css:11]: Use a css variable for color here

File: component/alert.css [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+}
+.alert-hr {
+  border: none;
+  border-top: 1px solid rgba(0,0,0,.125);
+}
+.alert__remove--btn {
+  position: absolute;
+  right: 0.5rem;
+  top: 0.5rem;
+  color: rgba(0,0,0,.6);
```
Comment (DanishOnGit) [inline component/alert.css:21]: use color variable

File: component/avatar.css [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+.avatar-sm {
+  width: 24px;
+  border-radius: 50%;
+  vertical-align: middle;
+}
+.avatar-xs {
+  width: 16px;
+  border-radius: 50%;
+  vertical-align: middle;
+}
```
Comment (DanishOnGit) [inline component/avatar.css:27]:
```suggestion
.avatar-xl, .avatar-sm,.avatar-xs,.avatar-default,.avatar-lg {
  border-radius: 50%;
  vertical-align: middle;
}

```
DRY code. You can set the width individually , but club other common properties as above

File: component/badge.on.avatar.css [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  transform: translate(50%, 50%);
+  /*  (x-axis(from left), y-axis(from top))  */
+  width: 1rem;
+  height: 1rem;
+  border-radius: 50%;
+  box-shadow: 0 0 0 0.15rem #fff;
+}
+
+.badge-avatar > .online {
+  background: #90eea8;
```
Comment (DanishOnGit) [inline component/badge.on.avatar.css:18]: use color varibales

File: component/alert.css (hunk 1 above)
Comment (Ros-an) [reply component/alert.css:3]: Ok, will do that.

File: component/alert.css (hunk 2 above)
Comment (Ros-an) [reply component/alert.css:15]: ya, i am working on it

File: component/button.css [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+    vertical-align: middle;
+    cursor: pointer;
+    width: 6.5rem;
+    border: 1px solid transparent;
+    padding: .375rem .75rem;
+    font-size: 1rem;
+    border-radius: .25rem;
+    font-family: inherit;
+    outline: none;
+}
```
Comment (DanishOnGit) [inline component/button.css:17]:
```suggestion
.btn {
    display: inline-block;
     font-family: inherit;
    font-weight: 400;
    font-size: 1rem;
    line-height: 1.5;
    text-align: center;
    text-decoration: none;
    vertical-align: middle;
    cursor: pointer;
    width: 6.5rem;
    border: 1px solid transparent;
    border-radius: .25rem;
     outline: none;
    padding: .375rem .75rem;
    
}
```
just a suggestion, you can write related properties together.

File: component/avatar.css (hunk 5 above)
Comment (Ros-an) [reply component/avatar.css:27]: Will do that.

File: component/button.css [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+.btn-danger {
+    color: #fff;
+    background-color: #dc3545;
+    border-color: #dc3545;
+}
+.btn-danger:hover {
+    color: #fff;
+    background-color: #bb2d3b;
+    border-color: #b02a37;
+}
```
Comment (DanishOnGit) [inline component/button.css:60]: please use color variables

Comment (Ros-an) [reply component/button.css:60]: working on it

File: main.css [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+@import url("component/cards.css");
+@import url("component/responsive.img.css");
+@import url("component/input.css");
+@import url("component/list.css");
+@import url("component/navbar.css");
+@import url("component/modal.css");
+@import url("component/rating.css");
+@import url("component/grid.css");
+@import url("component/toast.css");
+/* ------------------------------------- */
```
Comment (DanishOnGit) [inline main.css:25]: you can arrange these imports alphabetically to ease search

Comment (Ros-an) [reply main.css:25]: thats great, will work on all your suggestion. Thank you

File: main.css [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+  .header {
+    height: 80px;
+  }
+  .header > .heading {
+    font-size: 2.25rem;
+    padding-left: 0.5rem;
+  }
+  .header > .header-logo {
+    font-size: 1rem;
+  }
```
Comment (DanishOnGit) [inline main.css:125]:
```suggestion
  .header  .header-logo {
    font-size: 1rem;
  }
```
This should too work fine. Do let me know!

**Review by DanishOnGit — COMMENTED** [review]
Hi Roshan ,
Suggested changes are  around, using CSS variables, avoiding absolute units as far as possible, avoiding too complex selectors, DRY code, basic refactoring.

File: main.css (hunk 10 above)
Comment (Ros-an) [reply main.css:125]: Ya that will also work

**Review by Joyan11 — COMMENTED** [review]
Hello, Overall everything is done nicely. The classes are named well .
i also learned about the vertical align property which is something i had never used
I've suggested some suggested some minor changes for the outline property, refer this website http://www.outlinenone.com/ .
Good work.

**Comment by theniteshnarang** [comment]
Hey, @Ros-an , have you worked on the danish changes? Because I am not able to see it. If yes please make a new pull request for latest changes.

**Pull request merged**